Update README.md with troubleshooting instructions (#2751)
centerofci/mathesar: README.md

## Contributing
We actively encourage contribution! Read through [our contributing guidelines](https://wiki.mathesar.org/community/contributing) to get started.

## Bugs and troubleshooting

If you run into problems, here's what you can try:
1. Narrow down the problem you're having to a minimal search phrase
2. Search on [our issue tracker](https://github.com/centerofci/mathesar/issues/)
3. Search on [StackOverflow using our tag `[mathesar]`](https://stackoverflow.com/questions/tagged/mathesar)

> 4. If you think this is a novel problem, ask on one of those platforms or [on Matrix](#join-our-community)

Reviewer: I think it might be better if the default goto action for novel problems is to open tickets instead of matrix chats.

Reviewer 2: I agree with @pavish, an issue is much preferable to a Matrix chat. 

Also, I think this belongs in our wiki contributing guide, not in the README. Our README serves a wide audience and the goals are to provide an introduction to the project and to link out to appropriate resources for various sub-groups (contributors, users, people evaluating us, etc.) If we put primary documentation here, it will end up becoming long and unwieldy.

Author: > If you think this is a novel problem, ask on one of those platforms or on Matrix

This does instruct asking on the other platforms first and Matrix second. We can uncollapse "those platforms" so that it's something like:

```suggestion
4. If you think this is a novel problem, ask on [our issue tracker](https://github.com/centerofci/mathesar/issues/), [StackOverflow](https://stackoverflow.com/questions/tagged/mathesar), or [on Matrix](#join-our-community)
```

Another option could be to split asking on Matrix as a 5th step that says something like:
```suggestion
4. If you think this is a novel problem, ask on [our issue tracker](https://github.com/centerofci/mathesar/issues/) or [StackOverflow](https://stackoverflow.com/questions/tagged/mathesar)
5. In case you're having trouble formulating a report, reach out [on Matrix](#join-our-community)(#join-our-community)
```

@pavish @kgodey how do these sound?

@kgodey putting this on the wiki makes sense.

Reviewer 2: I like the second option with the 5th step. 

Also, do we actually want to be monitoring Stack Overflow? I'd consider removing that in favor of just the issue tracker. I don't have a strong opinion, though.

Author: @kgodey I don't have a strong opinion either; I think SO has its uses, and I expect people to increasingly use it, but we probably shouldn't actively encourage people to post on it at this time :+1:.
